WordSearch.java

```diff
@@ -0,0 +1,45 @@
+//Time Complexity O(3^n) 
+//Space Complexity O(length of word)
+class WordSearch{
```

Comment:
Review the complexities. I guess time should be size of matrix * 4^(length of word)